### PR TITLE
feat: derive serde for OpCode, improve implementations

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

File: crates/interpreter/src/instructions/opcode.rs [hunk 1]
```diff
@@ -463,13 +470,8 @@ impl OpCode {
...
 
     /// Instantiate a new opcode from a u8 without checking if it is valid.
-    ///
-    /// # Safety
-    ///
-    /// All code using `Opcode` values assume that they are valid opcodes, so providing an invalid
-    /// opcode may cause undefined behavior.
     #[inline]
-    pub unsafe fn new_unchecked(opcode: u8) -> Self {
+    pub fn new_unchecked(opcode: u8) -> Self {
```
Comment (rakita) [inline crates/interpreter/src/instructions/opcode.rs:474]: Made it unsafe as OpCode should not be possible to have an undefined byte

**Comment by rakita** [comment]
I have refactored this file in EOF branch, removed deprecated things, etc. if serde is needed lets make just that change and leave other changes to that branch.

File: crates/interpreter/src/instructions/opcode.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/interpreter/src/instructions/opcode.rs:474]: It's not memory unsafe and I'd like to use this to format any opcode, like `("{}",  Opcode::new_unchecked(op_byte))`

**Comment by DaniPopes** [comment]
Serde and formatting are nice to have for general use

**Review by mattsse — APPROVED** [review]
lgtm,

perhaps we limit this pr to serde only if @rakita prefers this

File: crates/interpreter/src/instructions/opcode.rs (hunk 1 above)
Comment (rakita) [reply crates/interpreter/src/instructions/opcode.rs:474]: It is better to not have undefined byte in struct

**Comment by rakita** [comment]
> Serde and formatting are nice to have for general use

> lgtm,
> 
> perhaps we limit this pr to serde only if @rakita prefers this

We can limit it to serde only.

**Review by rakita — APPROVED** [review]

**Pull request merged**